poe-tasks/publish-python-registry.sh

```diff
@@ -0,0 +1,218 @@
+    exit 0
+fi
+
+# Get package metadata from metadata.yaml
+PACKAGE_NAME=$(get_pypi_package_name "$METADATA_FILE")
+if [[ -z "$PACKAGE_NAME" ]]; then
+    echo "⚠️ Error: Package name not found in metadata.yaml, skipping PyPI publishing." >&2
+    exit 0
+fi
+BASE_VERSION=$(poe -qq get-version)
+
+# Determine version to use
+if [[ -n "$VERSION_OVERRIDE" ]]; then
+    VERSION="$VERSION_OVERRIDE"
+elif [[ "$PRE_RELEASE" == "true" ]]; then
+    # Add current timestamp for pre-release.
+    # we can't use the git revision because not all python registries allow local version identifiers. 
+    # Public version identifiers must conform to PEP 440 and only allow digits.
+    TIMESTAMP=$(date +"%Y%m%d%H%M")
+    VERSION="${BASE_VERSION}.dev.${TIMESTAMP}"
+else
+    VERSION="$BASE_VERSION"
+fi
+
+echo "Package name: $PACKAGE_NAME"
+echo "Version: $VERSION"
+
+# Check if package already exists
+if [[ $(curl -s -o /dev/null -w "%{http_code}" "$REGISTRY_CHECK_URL/$PACKAGE_NAME/$VERSION/json") == "200" ]]; then
+    echo "⚠️ Package $PACKAGE_NAME version $VERSION already exists on PyPI  at $REGISTRY_CHECK_URL/$PACKAGE_NAME/$VERSION/json. Skipping publishing."
+    exit 0
+else
+    echo "✅ Package $PACKAGE_NAME version $VERSION does not exist already on PyPI. Proceeding with publishing."
+fi
+
+
+# Assumes the connector uses Poetry for packaging and has a pyproject.toml
+if [[ -f "pyproject.toml" ]]; then
+    echo "Detected Poetry project"
+
+    VENV_DIR=.build-publish-venv
+    POETRY_VERSION=2.1.3
+
+    # runs automatically on script error or exit
+    cleanup() {
+        mv pyproject.toml.bak pyproject.toml
+        echo "Restored original pyproject.toml"
+    }
+    trap cleanup EXIT   
+
+    # to support overriding the package name and the version when publishing to PyPI, the script modifies the pyproject.toml file 
+    # we keep a backup at pyproject.toml.bak that is used to restore the initial state at the end
+    # TODO: figure out if we can do this in a less hacky way and reevaluate whether to continue defining PyPI package information in metadata.yaml
+    sed -i.bak -E \
+        "s/^([[:space:]]*name[[:space:]]*=[[:space:]]*\").*(\".*)$/\\1${PACKAGE_NAME}\\2/;
+        s/^([[:space:]]*version[[:space:]]*=[[:space:]]*\").*(\".*)$/\\1${VERSION}\\2/" \
+        pyproject.toml
+
+    echo "✅ Temporary override package name to '$PACKAGE_NAME' and version to '$VERSION' in pyproject.toml"
+
+    # Install dependencies
+    poetry install --all-extras
+
+    # Configure Poetry for PyPI publishing
+    poetry config repositories.mypypi "$REGISTRY_UPLOAD_URL"
+    poetry config pypi-token.mypypi "$PYPI_TOKEN"
+
+    # Default timeout is set to 15 seconds
+    # We sometime face 443 HTTP read timeout responses from PyPi
+    # Setting it to 60 seconds to avoid transient publish failures
+    export POETRY_REQUESTS_TIMEOUT=60
+
+    poetry publish --build --repository mypypi --no-interaction -vvv
```

Comment:
nonblocking: does this work?
```suggestion
    POETRY_REQUESTS_TIMEOUT=60 poetry publish --build --repository mypypi --no-interaction -vvv
```

Reply:
nice! I didn't know about this syntax. Seems like what we want